### PR TITLE
codegen: fix the run order of resolvers

### PR DESCRIPTION
**Pull request by giautm**
We (entgql) have a report from @amrnt about the unexpected result when he want to trying to use the custom Input field to set the value for the mutation's input.

Because with `range` the order of the map's iteration is randomized, so the custom field may run first then overrided by the original input.

You can run the test here few times without cache: https://github.com/ent/contrib/compare/master...amrnt:contrib:custom-input-field

I have:
 - [x] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [x] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

File: codegen/input.gotpl [hunk 1]
```diff
@@ -14,7 +14,16 @@
 			{{- end}}
 		{{- end }}
 
-		for k, v := range asMap {
+		fieldsInOrder := []string{
+			{{- range $field := .Fields }}
+				{{$field.Name|quote}},
+			{{- end }}
+		}
```
Comment (a8m) [inline codegen/input.gotpl:21]:
```suggestion
		fieldsInOrder := [...]string{
			{{- range $field := .Fields }}
				{{quote $field.Name}},
			{{- end }}
		}
```

Comment (a8m) [reply codegen/input.gotpl:21]: I also think newline is not needed and you can write it as:

```
fieldsInOrder := [...]string{ {{range .Fields}}{{quote .Name}},{{end}} }
```

**Review by a8m — APPROVED** [review]
Looks good.

**Comment by giautm** [comment]
Hello @StevenACoffman, please review this PR. I have added tests and updated the template to generate the input resolver

**Review by a8m — APPROVED** [review]
OK! Looks good 😍 

Thanks for addressing this @giautm

**Pull request merged**